src/util/splay_tree.ts

```diff
@@ -262,10 +272,18 @@ export class SplayTree<V> {
       const maxNode = leftTree.getMaximum();
       leftTree.splayNode(maxNode);
       leftTree.root.setRight(rightTree.root);
+      if (rightTree.root) {
+        rightTree.root.setParent(leftTree.root);
+      }
```

Comment:
```suggestion
      if (rightTree.root) {
        leftTree.root.setRight(rightTree.root);
        rightTree.root.setParent(leftTree.root);
      }
```
How about this?

Reply:
Thanks for the comments.

This is the intended behavior. Even if the `rightTree.root` is `undefined`, `right` of the `leftTree.root` must be set to `undefined`.